File: library/src/main/java/me/proxer/library/api/notifications/NotificationsApi.java
package me.proxer.library.api.notifications;

import retrofit2.Retrofit;

/**
 * API for the Notifications class.
 *
 * @author Ruben Gees
 */
public final class NotificationsApi {

    private final InternalApi internalApi;

    /**
     * Only for internal use.
     */
    public NotificationsApi(final Retrofit retrofit) {
        this.internalApi = retrofit.create(InternalApi.class);
    }

    /**
     * Returns the respective endpoint.
     */
    public NewsEndpoint news() {
        return new NewsEndpoint(internalApi);
    }

    /**
     * Returns the respective endpoint.
     */
    public NotificationsEndpoint notifications() {
        return new NotificationsEndpoint(internalApi);
    }

    /**
     * Returns the respective endpoint.
     */
    public NotificationInfoEndpoint notificationInfo() {
        return new NotificationInfoEndpoint(internalApi);
    }

    /**
     * Returns the respective endpoint.
     */
    public DeleteNotificationEndpoint deleteNotification(final String id) {
        return new DeleteNotificationEndpoint(internalApi, id);
    }

    /**
     * Returns the respective endpoint.
     */
    public DeleteNotificationEndpoint deleteAllNotifications() {
        return new DeleteNotificationEndpoint(internalApi, "0");
    }
}

File: library/src/main/kotlin/me/proxer/library/api/notifications/NotificationsApi.kt
package me.proxer.library.api.notifications

import retrofit2.Retrofit

/**
 * API for the Notifications class.
 *
 * @author Ruben Gees
 */
class NotificationsApi internal constructor(retrofit: Retrofit) {

    private val internalApi = retrofit.create(InternalApi::class.java)

    /**
     * Returns the respective endpoint.
     */
    fun news(): NewsEndpoint {
        return NewsEndpoint(internalApi)
    }

    /**
     * Returns the respective endpoint.
     */
    fun notifications(): NotificationsEndpoint {
        return NotificationsEndpoint(internalApi)
    }

    /**
     * Returns the respective endpoint.
     */
    fun notificationInfo(): NotificationInfoEndpoint {
        return NotificationInfoEndpoint(internalApi)
    }

    /**
     * Returns the respective endpoint.
     */
    fun deleteNotification(id: String): DeleteNotificationEndpoint {
        return DeleteNotificationEndpoint(internalApi, id)
    }

    /**
     * Returns the respective endpoint.
     */
    fun deleteAllNotifications(): DeleteNotificationEndpoint {
        return DeleteNotificationEndpoint(internalApi, "0")
    }
}
